docs/handlers.md

```diff
@@ -0,0 +1,118 @@
+# Handler plugin development
+
+The purpose of Handler plugins is to generate some kind of output from log entries.
+This can mean writing some representation of the entries into a log file, or sending
+the raw entries to an external service for further processing, or anything else, really.
+
+## The `HandlerPlugin` interface
+
+Each Handler plugin must implement the `HandlerPlugin` interface, which extends
+the generic [`Plugin` interface] and adds the following methods and properties:
+
+ - `public log(entry: LogEntry): Promise<void> | void`: This method will be called from
+   `Logger.add()`. Whatever a handler wants to do with log entries is up to the handler.
+ - `public runTask?<R>(cb: () => R): R`: This optional method can be implemented by
+   handlers which need to integrate with Debugr tasks, e.g. to wrap the task execution
+   in their internal `AsyncLocalStorage` instance.
+
+## Entry snapshots
+
+One thing worth noting is that the call chain all the way from the public `Logger` API
+methods through the semi-internal `add()` method down to the handlers' `log()` method is
+_synchronous_, so whatever the handlers do _synchronously_ (until the first `await`)
+within `log()` should be _fast_, otherwise it will slow down code executing at the calling
+site. This is intentional on the part of Debugr: this way we can _guarantee_ that any mutable
+data passed to Debugr will not be altered by user-land code before handlers get a chance
+to process the data, or take a snapshot to be processed later. Consider the following:
+
+```typescript
+const user = await em.find(User, 'some id');
```

Comment:
Nezrejme co je to "em" (jo, ja to vim, ale je to nezrejme). Co misto "em" dat "databaseManager", coz je genericky dementni nazev, ale da se pod tim neco vic predstavit :)

Reply:
`em` je imo dost jasny, ale ok, dame `db`